### PR TITLE
Feature/134 addfeatured image for guideline pages

### PR DESCRIPTION
**Pull request by MerlindlH**
Issue:
https://github.wdf.sap.corp/gd/FioriExperience/issues/134

TODOs left:
- [x] add proper images to the existing posts
- [x] give images a fixed size on the Controls Page
- [ ] show images on existing posts

### CODE REVIEW

File: static/admin/config.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ collections:
     fields:
       - {label: "Template Key", name: "templateKey", widget: "hidden", default: "design-guideline-post"}
       - {label: "Title", name: "title", widget: "string"}
+      - {label: "Featured Image", name: "featuredImage", widget: "image", tagname: ''}
```
Comment (devinea) [inline static/admin/config.yml]: Should add these as required attributes.

Comment (MerlindlH) [reply static/admin/config.yml]: If I have seen this right, required is true by default (at least I couldn't save the page without selecting an image), but I'll check this again

Comment (MerlindlH) [reply static/admin/config.yml]: image
from [here](https://www.netlifycms.org/docs/configuration-options/), required defaults to true

Comment (devinea) [reply static/admin/config.yml]: Great. Thanks 👍

Comment (MerlindlH) [reply static/admin/config.yml]: https://github.com/devinea/mvx19-cms/pull/26/files#diff-e36160419abb71f07eb8f74e9d3cc8bcL21 do you want this to be required as well? (the featuredImage in the blog)

**Comment by MerlindlH** [comment]
I've just seen there are gatsby `Img` tags as well, shall I replace my html-`img` tags with the gatsby ones?

**Comment by devinea** [comment]
@MerlindlH - yes we should use `Img` by default for images. This allows the images to be optimised using `gatsby-image` during the build.

File: src/pages/design/controls.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import Layout from '../../components/Layout'
+import GuidelineHeader from '../../components/guideline-header/guideline-header'
+import GuidelineLeftNav from '../../components/guideline-left-nav/guideline-left-nav';
+import './controls.scss'
+import { graphql, Link, StaticQuery } from "gatsby";
+import Img from 'gatsby-image';
+
+class DesignControlsIndexPage extends React.Component {
```
Comment (devinea) [inline src/pages/design/controls.js]: I think longer term we should make this into a component.
Taking out the following and moving to `design/guideline.js`
```
<Layout>
        <GuidelineHeader></GuidelineHeader>
        <GuidelineLeftNav></GuidelineLeftNav>
```

File: static/admin/config.yml (hunk 1 above)
Comment (devinea) [reply static/admin/config.yml]: Yes.

File: src/pages/design/controls.js [hunk 3]
```diff
@@ -8,22 +8,22 @@ import Img from 'gatsby-image';
...
-              <Link className="control-menu" key={data.id} to={data.fields.slug}>
-                <Img className="control-img" imgStyle={{ 'object-fit': 'contain' }}
+              <Link className="control-menu" key={data.id} to={data.fields.slug} >
+                <Img className="control-img" imgStyle={{ 'objectFit': 'contain' }}
                      sizes={data.frontmatter.featuredImage.childImageSharp.sizes}/>
                 <div className="control-title">{data.frontmatter.title}</div>
                 </Link>
             ))}
-        </section>
+        </secton>
```
[the remark this answers is not in the document]
Comment (devinea) [reply src/pages/design/controls.js]: oops

File: src/pages/design/controls.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
+.design-controls {
```
Comment (devinea) [inline src/pages/design/controls.scss]: Can you remove scss?

Comment (MerlindlH) [reply src/pages/design/controls.scss]: sure, let me try :)

Comment (MerlindlH) [reply src/pages/design/controls.scss]: Or maybe later on as you suggested :D

File: static/admin/config.yml (hunk 1 above)
Comment (MerlindlH) [reply static/admin/config.yml]: Okay, will be adjusted :)

**Pull request merged**